src/channel/rpc_0_7_1.ts

```diff
@@ -180,25 +196,37 @@ export class RpcChannel {
   }
 
   /**
-   * fetch if undefined else just return this.specVersion
-   * return this.specVersion as 'M.m'
-   * @example this.specVersion = "0.8"
+   * fetch rpc node specVersion
+   * @example this.specVersion = "0.7.1"
    */
-  public async getSpecVersion() {
-    if (!this.specVersion) {
-      const extendedVersion = await this.getSpecificationVersion();
-      const [major, minor] = extendedVersion.split('.');
-      const specVerson = `${major}.${minor}` as SupportedRpcVersion;
-      this.specVersion ??= specVerson;
-    }
-    return this.specVersion;
+  public getSpecVersion() {
+    return this.fetchEndpoint('starknet_specVersion');
   }
 
   /**
-   * fetch spec version in extended format "M.m.p-?"
+   * fetch if undefined test and set specVersion, else just return this.specVersion
+   * @example this.specVersion = "0.7.1"
    */
-  public getSpecificationVersion() {
-    return this.fetchEndpoint('starknet_specVersion');
+  public async setupSpecVersion() {
```

Comment:
Nitpick, `setup` is usually used as a noun and `set up` as the corresponding verb form so the method would be `setUpSpecVersion`.